core/src/test/java/org/infinispan/dataconversion/Gzip.java

```diff
@@ -0,0 +1,37 @@
+package org.infinispan.dataconversion;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+public final class Gzip {
+
+   public static byte[] compress(String str) {
+      try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
We should probably use `ExposedByteArrayOutputStream` in both of these places to avoid copying the byte[].

Reply:
Sorry, I know this code didn't use it before.

Reply:
and it's also a test

Reply:
Ah yes I didn't notice.